QuestionBot/QuestionBot/Model/MessageEmitter.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace QuestionBot.Model {
+    public class MessageEmitter : IMessageEmitter {
+        private string _lineInput = "";
+        private static IList<IMessageListener> _listeners = new List<IMessageListener>();
```

Comment:
There are tradeoffs here but they are not performance related, they are design related.

Right now, `MessageEmitter` is basically a static class. By this I mean that an instance of MessageEmitter (ie `new MessageEmitter()`) essentially holds no data because all of the data (in this case, just `_listeners`) is static. So all instances of MessageEmitter have to be the same (MessageEmitter a **can't** have different listeners than MessageEmitter b).

However, if we make `_listeners` an instance variable, then each instance of `MessageEmitter` can have its own listeners.

So I think we have 2 paths forward here. Either we keep `_listeners` as static and make the whole class explicitly static, or we make `_listeners` an instance member. What do you think?